### PR TITLE
ss/daily filter

### PR DESCRIPTION
**Pull request by ssiegelx**
This contains all of the modifications to draco needed to implement the daily foreground filtering pipeline, which beamforms a Timestream along north-south baselines and then applies a high-pass filter along the frequency axis.  It primarily updates various tasks to work with the HybridVisStream.

### CODE REVIEW

**Comment by ssiegelx** [comment]
Apologies for putting this all in a single PR.  We probably want to hold off on merging this until we have tested on the 100 days and confirmed that it works.

**Review by ljgray — APPROVED** [review]
I've gone through the changes here and I think it all looks good, other than a couple of very minor comments. It would be nice to get this branch merged in since everything here seems to be more or less complete and tested, and any extra changes we end up needing could be a separate PR to make it easier to review.

The commits should probably be cleaned up with an interactive rebase before merging.

**Comment by ljgray** [comment]
Also, there are a few places where the `ruff` linting is failing

File: draco/analysis/ringmapmaker.py
Comment (ssiegelx) [reply draco/analysis/ringmapmaker.py:116]: The SiderealGrouper task keeps a small amount of data that extends beyond the boundaries of the sidereal day. If you convert to RA here, then the RA axis will wrap, which causes issues later on when passed to the SiderealRebinner task.  This line sets the timestamps before the sidereal day to RAs less than 0 deg and timestamps after the sidereal day to RAs greater than 360 degrees.  This behaves correctly when passed to SiderealRebinner.

File: draco/analysis/dayenu.py
Comment (Arnab-half-blood-prince) [inline draco/analysis/dayenu.py:413]: tau_centre should be tauc

File: draco/analysis/dayenu.py
Comment (Arnab-half-blood-prince) [inline draco/analysis/dayenu.py:411]: tau_width should be tauw

File: draco/analysis/flagging.py
Comment (Arnab-half-blood-prince) [inline draco/analysis/flagging.py:1694]: Is rho would be `rho = self.rho` ?

**Review by Arnab-half-blood-prince — COMMENTED** [review]
looks good to me. I just have only a few comments. Most of these new tasks, I checked and it is working.

File: draco/analysis/flagging.py
Comment (ssiegelx) [reply draco/analysis/flagging.py:1694]: Rho determines how the threshold is lowered with each iteration and is used to determine self.threshold in the setup method.  For chi-squared masking and sensitivity masking, I've moved this iteration outside of the sumthreshold function so that we can also update the median and median absolute deviation as we lower the threshold.

**Review by Arnab-half-blood-prince — APPROVED** [review]

**Pull request merged**
